lib/datadog/statsd.rb

```diff
@@ -432,17 +434,28 @@ def send_stats(stat, delta, type, opts=EMPTY_OPTIONS)
 
     def send_stat(message)
       if @batch_nesting_depth > 0
+        message_bytes = message.bytesize
```

Comment:
I think this leaves open the (unlikely) possibility of `message_bytes` being greater than `@max_buffer_size` and us ending up with the same problem. What do you think about raising an error if that ever happens? 

```ruby
if message_bytes >= @max_buffer_bytes
  raise ArgumentError.new("Message size is too large.")
end
```